File: src/main/java/com/cloudpi/cloudpi_backend/files_info/entities/java/FilesystemObjectEntity.java
package com.cloudpi.cloudpi_backend.files_info.entities.java;

import com.cloudpi.cloudpi_backend.files_info.entities.kotlin.RootDirectoryEntity;

import javax.persistence.*;
import java.util.Date;

@Entity
@DiscriminatorValue(value = "FS_OBJECT")
public class FilesystemObjectEntity extends FilesystemIdEntity {
    @Column(nullable = false)
    private String name;
    @Column(unique = true, nullable = false)
    private String relativePath;
    @ManyToOne
    @JoinColumn(name = "parent_id", nullable = false)
    private FilesystemIdEntity parent;
    @ManyToOne
    @JoinColumn(name = "root_id", nullable = false)
    private RootDirectoryEntity root;
    @Column(nullable = false, updatable = false)
    private Date createdAt;
}

File: src/main/java/com/cloudpi/cloudpi_backend/files_info/entities/kotlin/FilesystemObjectEntity.kt
package com.cloudpi.cloudpi_backend.files_info.entities.kotlin

import com.cloudpi.cloudpi_backend.user.entities.UserEntity
import java.util.*
import javax.persistence.*

//@Entity
//@DiscriminatorValue(value = "FS_OBJECT")
abstract class FilesystemObjectEntity(
    id: Long,
    owner: UserEntity,
    permissions: List<FilePermissionEntity>,

    @Column(nullable = false)
    open var name: String,
    @Column(unique = true, nullable = false)
    open var relativePath: String,
    @ManyToOne
    @JoinColumn(name = "parent_id", nullable = false)
    open var parent: FilesystemIdEntity,
    @ManyToOne
    @JoinColumn(name = "root_id", nullable = false)
    open var root: RootDirectoryEntity,
    @Column(nullable = false, updatable = false)
    open var createdAt: Date,


    ) : FilesystemIdEntity(
    id,
    owner,
    permissions
)